examples/async_pcb_insert_drq/async_drq_randomized.py

```diff
@@ -88,13 +91,43 @@ def print_green(x):
     return print("\033[92m {}\033[00m".format(x))
 
 
+SHOULD_PAUSE = False  # flag to pause actor/learner agents
+
+
+def pause_client(a, b):
+    """Set flag to pause actor/learner agents at next iteration"""
+    global SHOULD_PAUSE
+    SHOULD_PAUSE = True
+    print("Requested pause training")
+
+
+def on_press(key):
+    """Callback for when a key is pressed"""
+    try:
+        # print(f'{key.char} pressed')
+
+        # chosen a rarely used key to avoid conflicts. this listener is always on, even when the program is not in focus
+        if key == pynput.keyboard.Key.pause:
+            print("Pause pressed")
+            pause_client(None, None)
+    except AttributeError:
+        # print(f'{key} pressed')
+        pass
+
+
+signal.signal(signal.SIGUSR1, pause_client)  # enable interrupt signal to pause training
+listener = pynput.keyboard.Listener(on_press=on_press)  # to enable keyboard based pause
```

Comment:
I didnt try it on my pc. can we just have a single keyboard listener for this. It seems that we can simplify the code without having both `pause_client` and `on_press` fn. It would be good to just have a single `on_press_pause` function. 

Also, what is the exact keyboard key to pause this operation?

Reply:
Sure yeah we can reduce down to one listener and one callback to handle the event. I still do not know of a way to make it react to keypresses *only* when the program is in the foreground. This works if you hit the right keys at anytime.

@ key: It is the 'pause break' key. People can also define their own key of course, or use a function key.

Reply:
reduced impl to one keyboard event listener and one callback `pause_callback()`